### ISSUE TITLE
download data

### ISSUE THREAD

**Issue by sws827767542** [wwangweii/CcSi_MHAHGEL#1]
<img width="301" alt="Image" src="https://github.com/user-attachments/assets/d419d314-7390-408d-ab80-ee2b058d5b4b" />Hello, I have now reproduced your code. However, the performance is only 65%. Could you please advise which part of the code I might need to adjust, such as the parameters? I look forward to your response. Thank you.

**Comment by wwangweii** [comment]
Are you working with the entire ABIDE dataset? If so, replicate the experiments on the four sites described in the paper. Additionally, test multiple dataset split seeds—for example, 50 seeds using state = [i for i in range(50)]—and report the average accuracy. There might be outliers where the model underperforms, but the average results should align with the paper's results.

If you plan to evaluate on the full ABIDE dataset, ensure your code supports processing data across three brain atlases (you'll need to modify the code accordingly). Using three atlases significantly outperforms two.

**Comment by Jane-YU3** [comment]
> <img alt="Image" width="301" src="https://private-user-images.githubusercontent.com/149987355/447009343-d419d314-7390-408d-ab80-ee2b058d5b4b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg1Mzk0MjIsIm5iZiI6MTc1ODUzOTEyMiwicGF0aCI6Ii8xNDk5ODczNTUvNDQ3MDA5MzQzLWQ0MTlkMzE0LTczOTAtNDA4ZC1hYjgwLWVlMmIwNThkNWI0Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyMlQxMTA1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Mjc0MmE0Njc2YjA4NGI1ZWZjYmFkODcxMTExMzA0ZDE5MjYxYTE5ZmNlZWYxOWNjOGMwODU1NTUwZTU5MDc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.K0qOulaEyZgr28-6dyxy-mCWto_tH9uDU9ATsscwh7k">Hello, I have now reproduced your code. However, the performance is only 65%. Could you please advise which part of the code I might need to adjust, such as the parameters? I look forward to your response. Thank you.

Hello, I'm also trying to reproduce this code. Have you improved the accuracy? If so, could you share your experience with me? Thank you very much!

**Comment by wwangweii** [comment]
> > <img alt="Image" width="301" src="https://private-user-images.githubusercontent.com/149987355/447009343-d419d314-7390-408d-ab80-ee2b058d5b4b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg1Mzk0MjIsIm5iZiI6MTc1ODUzOTEyMiwicGF0aCI6Ii8xNDk5ODczNTUvNDQ3MDA5MzQzLWQ0MTlkMzE0LTczOTAtNDA4ZC1hYjgwLWVlMmIwNThkNWI0Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyMlQxMTA1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Mjc0MmE0Njc2YjA4NGI1ZWZjYmFkODcxMTExMzA0ZDE5MjYxYTE5ZmNlZWYxOWNjOGMwODU1NTUwZTU5MDc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.K0qOulaEyZgr28-6dyxy-mCWto_tH9uDU9ATsscwh7k">Hello, I have now reproduced your code. However, the performance is only 65%. Could you please advise which part of the code I might need to adjust, such as the parameters? I look forward to your response. Thank you.
> 
> Hello, I'm also trying to reproduce this code. Have you improved the accuracy? If so, could you share your experience with me? Thank you very much!

Sorry for missing this message. Here are some details regarding the code:  
1) Check the original BOLD signals, input FCN, and input H to ensure their values are correct;  
2) Verify the atlas weights for each atlas. As specified in the paper, the weight is 0.7 for the HO atlas and 0.3 for the AAL atlas;  
3) Confirm the selected four sites in the code, ensuring they align with what is reported in the paper;  
4) Test more states (see Line 321 in main.py) — for example, use `state = [i for i in range(50)]` — and select appropriate values for learning rate (lr) and maximum epochs (max-epoch).

**Comment by 0neNoth1ng** [comment]
<img width="888" height="217" alt="Image" src="https://github.com/user-attachments/assets/66d743e8-84b0-447d-bcf4-cd213a092d6e" />
我用的是Amazon下载的预处理数据集，filt_global，cpac，只用了AAL和HO两个脑区模板，被试数量为884个，运行完代码结果不是很理想，虽然论文和作者上面说到中说到，只用的四个特定的站点数据。如果只用四个站点，那结果应该确实会好，但是为什么不用全部的数据呢？

**Comment by wwangweii** [comment]
> <img alt="Image" width="888" height="217" src="https://private-user-images.githubusercontent.com/97232226/509294628-66d743e8-84b0-447d-bcf4-cd213a092d6e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI2MDYwODYsIm5iZiI6MTc2MjYwNTc4NiwicGF0aCI6Ii85NzIzMjIyNi81MDkyOTQ2MjgtNjZkNzQzZTgtODRiMC00NDdkLWJjZjQtY2QyMTNhMDkyZDZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTA4VDEyNDMwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwZTAyZGFkZWExYTQwYzY3ODI5OWNkNDc3ZDFmZDBiZjFlMDYyZmRlZTRjZmNhOTEwMGYyYWJjNjhmNzBiMmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dr4Q9smuTDVFlu6mKttTw8_jb27_JSZWnkwm_glWO40"> 我用的是Amazon下载的预处理数据集，filt_global，cpac，只用了AAL和HO两个脑区模板，被试数量为884个，运行完代码结果不是很理想，虽然论文和作者上面说到中说到，只用的四个特定的站点数据。如果只用四个站点，那结果应该确实会好，但是为什么不用全部的数据呢？

Thank you for your attention. First, there are several reasons for the performance on the entire ABIDE dataset: 1) The heterogeneity of large-scale multi-site data does affect the results, and our site consistency regularization may not fully overcome this impact; 2) The feature fusion of AAL and HO may fail to fully capture the feature patterns of the entire ABIDE data. 

Therefore, the results on the entire ABIDE dataset in our paper are based on three Atlases (detailed in the paper, with additional DOH features). Through parameter tuning (regularization parameters and atals feature weights) and increasing the number of 5/10-fold validation experiments, the experimental results will be consistent with those reported in the paper. We hope this helps you.

**Comment by 0neNoth1ng** [comment]
<img width="659" height="315" alt="Image" src="https://github.com/user-attachments/assets/e510e84c-fdd9-4e31-8d15-eb9f59e6b039" />

<img width="778" height="643" alt="Image" src="https://github.com/user-attachments/assets/67ee0c15-5d64-4812-b2a9-b0d50686ba51" />

您论文中使用了NYU、UM1、USM、YALE四个站点的数据，总共是355名被试，我将预处理的数据进行筛选，筛选出这四个站点的数据，同时去除了包含0值ROI的被试，共307个被试，理论上讲，效果应该会更好

<img width="1647" height="407" alt="Image" src="https://github.com/user-attachments/assets/eb6d5b24-be32-40d0-9acf-57883bb6b5a5" />

但是重新跑了一遍您的代码之后，效果任然不理想，平均测试准确率只有61%

这个数据集确实难跑，最后还有个问题想请教一下您，我自己的模型训练时，训练集Loss正常下降，准确率正常跑到99%，但是我的验证集Loss一直不下降，反而上升到1~2区间，准确率也一直反复波动，效果十分不理想，想请教一下您，我验证集的Loss值怎么样才能下降？

感谢您百忙之中抽出时间给我答复！不胜感激！

**Comment by wwangweii** [comment]
> <img alt="Image" width="659" height="315" src="https://private-user-images.githubusercontent.com/97232226/513170059-e510e84c-fdd9-4e31-8d15-eb9f59e6b039.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMwODgxODIsIm5iZiI6MTc2MzA4Nzg4MiwicGF0aCI6Ii85NzIzMjIyNi81MTMxNzAwNTktZTUxMGU4NGMtZmRkOS00ZTMxLThkMTUtZWI5ZjU5ZTZiMDM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE0VDAyMzgwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3YjEwNDY3NmY4N2IxNjJlODlhNjcwYzhkYThhZTM3MTUyYmU2OWJkYjRiMzVjMmNjOTZjOTcxMThkMjMyZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zSXoYW_XX69ZvM3v-pclqMvmZ85RyWno415xR_2b290"> <img alt="Image" width="778" height="643" src="https://private-user-images.githubusercontent.com/97232226/513171276-67ee0c15-5d64-4812-b2a9-b0d50686ba51.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMwODgxODIsIm5iZiI6MTc2MzA4Nzg4MiwicGF0aCI6Ii85NzIzMjIyNi81MTMxNzEyNzYtNjdlZTBjMTUtNWQ2NC00ODEyLWIyYTktYjBkNTA2ODZiYTUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE0VDAyMzgwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjYjc0MzZkNjQ2ZDk0NGM0NjNmOWFlNGNhMzMxYjRhNmRmZGI0MjgyYWEzNDJlMzgwZWMzZGY3YmE3OWVjZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WZ3LItS1VM9XMwwtC4hpnxNA0ltHYrAVJzZT6hILFqY">
> 您论文中使用了NYU、UM1、USM、YALE四个站点的数据，总共是355名被试，我将预处理的数据进行筛选，筛选出这四个站点的数据，同时去除了包含0值ROI的被试，共307个被试，理论上讲，效果应该会更好
> 
> <img alt="Image" width="1647" height="407" src="https://private-user-images.githubusercontent.com/97232226/513171693-eb6d5b24-be32-40d0-9acf-57883bb6b5a5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMwODgxODIsIm5iZiI6MTc2MzA4Nzg4MiwicGF0aCI6Ii85NzIzMjIyNi81MTMxNzE2OTMtZWI2ZDViMjQtYmUzMi00MGQwLTlhY2YtNTc4ODNiYjZiNWE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE0VDAyMzgwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1M2U0ODI2MmVmZWFlNTRjMzk0NDZjN2EzZTg3YTgwMjc2MzMyNDk1OWMxMmIwYTA5MTIwYzgyMGNjOTVkY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.K29y-eulL84PjFeAYEH5GrTXHBcIQSs37zS6ux_4ioM">
> 但是重新跑了一遍您的代码之后，效果任然不理想，平均测试准确率只有61%
> 
> 这个数据集确实难跑，最后还有个问题想请教一下您，我自己的模型训练时，训练集Loss正常下降，准确率正常跑到99%，但是我的验证集Loss一直不下降，反而上升到1~2区间，准确率也一直反复波动，效果十分不理想，想请教一下您，我验证集的Loss值怎么样才能下降？
> 
> 感谢您百忙之中抽出时间给我答复！不胜感激！


It is highly unexpected based on your results. First, check if there are any issues with the downloaded data and labels. The increase in the validation set performance is related to overfitting. If the problem remains unresolved, you can try contacting me via email at "kkwweii@163.com" for further communication.
